pkg/compactor/bucket_compactor.go

```diff
@@ -680,6 +699,7 @@ type BucketCompactor struct {
 	compactDir                     string
 	bkt                            objstore.Bucket
 	concurrency                    int
+	skipUnhealthyBlocks            bool
```

Comment:
I think it might make sense to have this new flag also control what `skipBlocksWithOutOfOrderChunks` currently does, and deprecated `skipBlocksWithOutOfOrderChunks`. In the end I think they would be used in the same way, blocks are marked as no-compact, and someone will have to check the marked blocks and try to repair, so it might make sense to just have one flag that covers all health issues a compactor might find with a block.

Reply:
I agree with Vernon to repurpose existing `skipBlocksWithOutOfOrderChunks` option for skipping all unhealthy blocks.